src/http/shim.rs

```diff
@@ -0,0 +1,46 @@
+//! Shim traits to define an external client as an unleash HTTP client This is
+//! an imperfect system, but sufficient to work with multiple async frameworks,
+//! which is the goal.
+
+// Copyright 2022 Cognite AS
+
+use core::fmt::{Debug, Display};
+use std::error::Error;
+
+use async_trait::async_trait;
+use serde::{de::DeserializeOwned, Serialize};
+
+/// Abstraction over the concrete HTTP client being used. Implement this on any
+/// type to use it as an HTTP client.
+#[async_trait]
+pub trait HttpClient: Sync + Send {
+    type HeaderName: Clone + Sync + Send;
+    type Error: Debug + Display + Error + Send + Sync + 'static;
+    type RequestBuilder;
+
+    /// Construct a HTTP client layer headername
+    fn build_header(name: &'static str) -> Result<Self::HeaderName, Self::Error>;
+
+    /// Make a get request
+    fn get(&self, uri: &str) -> Self::RequestBuilder;
+
+    /// Make a post  request
```

Comment:
```suggestion
    /// Make a post request
```

Reply:
Ah, didn't notice the auto merge, so this is now a moot point and really just a whitespace character. Marking this as resolved